code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -21,9 +21,19 @@
 
 /datum/robot_component/proc/install()
 /datum/robot_component/proc/uninstall()
+	//Integrated Lights removed? Lights out.
```

Comment:
uh, what? No. That's not even slightly okay. Overriding a proc that is intended to be empty, that's not okay at all. Override robot_component/integrated_light/uninstall if you want to make it turn off when uninstalled.